pkg/apis/core/v1alpha1/tiltrun_types.go

```diff
@@ -0,0 +1,274 @@
+		fieldErrors = append(fieldErrors, field.Required(field.NewPath("tiltfilePath"), "cannot be empty"))
+	}
+	validExitCondition := false
+	for _, v := range exitConditions {
+		if v == in.Spec.ExitCondition {
+			validExitCondition = true
+			break
+		}
+	}
+	if !validExitCondition {
+		var detailMsg strings.Builder
+		detailMsg.WriteString("valid values: ")
+		for i, ec := range exitConditions {
+			if i != 0 {
+				detailMsg.WriteString(", ")
+			}
+			detailMsg.WriteString(string(ec))
+		}
+
+		fieldErrors = append(fieldErrors, field.Invalid(
+			field.NewPath("exitCondition"),
+			in.Spec.ExitCondition,
+			detailMsg.String()))
+	}
+	return fieldErrors
+}
+
+var _ resource.ObjectList = &TiltRunList{}
+
+func (in *TiltRunList) GetListMeta() *metav1.ListMeta {
+	return &in.ListMeta
+}
+
+// TiltRunStatus defines the observed state of TiltRun
+type TiltRunStatus struct {
+	// PID is the process identifier for this instance of Tilt.
+	PID int64 `json:"pid"`
+	// StartTime is when the Tilt engine was first started and began processing resources.
+	StartTime metav1.MicroTime `json:"startTime"`
+	// Tiltfile is the current state of the Tilt engine including details about the last time the Tiltfile was
+	// processed and any build errors (e.g. syntax errors) encountered.
+	Tiltfile TiltfileState `json:"tiltfile"`
+	// Resources are normalized state representations of the servers/jobs managed by this TiltRun.
+	Resources []ResourceState `json:"resources"`
+
+	// Done indicates whether this TiltRun has completed its work and is ready to exit.
+	Done bool `json:"done"`
+	// Error is a non-empty string when the TiltRun is Done but encountered a failure as defined by the ExitCondition
+	// from the TiltRunSpec.
+	Error string `json:"error,omitempty"`
+}
+
+// TiltfileState includes details about the engine.
+type TiltfileState struct {
+	// Build includes the active Tiltfile build (if any) as well as the last build (if any). Pending is always nil, as
+	// Tiltfile changes are currently processed immediately.
+	Build BuildState
+}
+
+// ResourceState contains a normalized representation of build and runtime state for a resource managed by this TiltRun.
+type ResourceState struct {
+	// Name is the name of the resource, typically defined via a call to a resource function in the Tiltfile.
+	Name string `json:"name"`
+	// Build provides information about pending/active/terminated build(s) for the resource.
+	//
+	// If nil, the resource does not perform builds (for example, a local resource without a serve_cmd).
+	Build *BuildState `json:"build,omitempty"`
+	// Runtime provides information about the current execution of the resource.
+	Runtime RuntimeState `json:"runtime,omitempty"`
+}
+
+// BuildState includes details about a currently pending build, currently active build, and (last) terminated build.
+type BuildState struct {
+	// Pending gives details about the currently enqueued build (if any).
+	Pending *PendingBuild `json:"pending,omitempty"`
```

Comment:
i would probably use the convention `BuildStatePending`, to match the kubernetes naming convention 

https://pkg.go.dev/k8s.io/api/core/v1#ContainerStateWaiting